src/test/java/com/vfasad/test/service/OptionServiceTest.java

```diff
@@ -0,0 +1,71 @@
+package com.vfasad.test.service;
+
+import com.vfasad.entity.Option;
+import com.vfasad.repo.OptionRepository;
+import com.vfasad.service.OptionName;
+import com.vfasad.service.OptionService;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import static org.mockito.Mockito.*;
+import static org.junit.Assert.*;
+
+import java.util.*;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+@RunWith(MockitoJUnitRunner.class)
+public class OptionServiceTest {
+    @Mock
+    private OptionRepository optionRepository;
+
+    @InjectMocks
+    OptionService optionService;
+
+    @Test
+    public void testFindAll() {
+        List<Option> optionList = generateOptionList(false);
+        when(optionRepository.findAll()).thenReturn(new ArrayList<Option>(){{add(new Option());}});
+        LinkedHashMap<OptionName, Option> expectedResult =
+                optionList.stream().collect(Collectors.toMap(Option::getName, Function.identity(), (o1,o2) -> o2, LinkedHashMap::new));
+        expectedResult.put(null, new Option());
```

Comment:
Why do we put that 'null' option here?..

Reply:
I've mocked that optionRepository.findAll() returns List with the one new Option() (name and value are null) - row 31. In service method values from list that are returned from optionRepository.findAll() should be added to values from OptionName enum. So I want to check that additional Option will be added to Collection of Options.